cli/src/commands/graph/monograph/commands/update.ts

```diff
@@ -41,7 +50,31 @@ export default (opts: BaseCommandOptions) => {
       }
     }
 
-    const spinner = ora('Federated Graph is being updated...').start();
+    if (options.subscriptionProtocol && !isValidSubscriptionProtocol(options.subscriptionProtocol)) {
+      program.error(
+        pc.red(
+          pc.bold(
+            `The subscription protocol '${pc.bold(
+              options.subscriptionProtocol,
+            )}' is not valid. Please use one of the following: sse, sse_post, ws.`,
+          ),
+        ),
+      );
+    }
+
+    if (options.websocketSubprotocol && !isValidWebsocketSubprotocol(options.websocketSubprotocol)) {
```

Comment:
Why is this code duplicated?

Reply:
we have to check if the passed option is valid in each of the commands which take subprotocol as input